lib/blemulator/internal/simulation_manager_mixins/characteristics_mixin.dart

```diff
@@ -145,4 +147,79 @@ mixin CharacteristicsMixin on SimulationManagerBase {
       ));
     }
   }
+  Future<void> _monitorCharacteristicForIdentifier(
+    int characteristicIdentifier,
+    String transactionId,
+  ) async {
+    SimulatedCharacteristic targetCharacteristic =
```

Comment:
Where are the error checks? Change mixin to `... on SimulationManagerBaseWithErrorChecks`.

Reply:
This also pertains to other two methods